lib/streamlit/commands/execution_control.py

```diff
@@ -67,3 +70,19 @@ def rerun() -> None:
         )
         # Force a yield point so the runner can do the rerun
         st.empty()
+
+
+@gather_metrics("experimental_rerun")
+def experimental_rerun() -> None:
+    """Rerun the script immediately.
+
+    When `st.experimental_rerun()` is called, the script is halted - no
+    more statements will be run, and the script will be queued to re-run
+    from the top.
+    """
+    msg = make_deprecated_name_warning("experimental_rerun", "rerun", "2024-04-01")
```

Comment:
I chose April 2024 somewhat arbitrarily. 3 months felt short, esp with the lower visibility of the warning.